src/app/application.cpp

```diff
@@ -403,14 +403,12 @@ void Application::runExternalProgram(const BitTorrent::Torrent *torrent) const
     LogMsg(tr("Torrent: %1, running external program, command: %2").arg(torrent->name(), program));
 
 #if defined(Q_OS_WIN)
-    auto programWchar = std::make_unique<wchar_t[]>(program.length() + 1);
-    program.toWCharArray(programWchar.get());
-
     // Need to split arguments manually because QProcess::startDetached(QString)
     // will strip off empty parameters.
     // E.g. `python.exe "1" "" "3"` will become `python.exe "1" "3"`
     int argCount = 0;
-    std::unique_ptr<LPWSTR[], decltype(&::LocalFree)> args {::CommandLineToArgvW(programWchar.get(), &argCount), ::LocalFree};
+    std::unique_ptr<LPWSTR[], decltype(&::LocalFree)> args {::CommandLineToArgvW(
+        reinterpret_cast<LPCWSTR>(program.utf16()), &argCount), ::LocalFree};
```

Comment:
Seems it isn't a really good idea. 
https://stackoverflow.com/questions/23124781/what-is-the-fastest-method-to-get-const-wchar-t-from-qstring
Even if by a lucky chance it works, it cannot be considered other than as a workaround. So I have a question, does the affected code really need to improve performance in this way?

Reply:
OK, I updated the commit.

>So I have a question, does the affected code really need to improve performance in this way?

I have little concern of performance for the commit, it is more about unifying coding style and using the proper conversion method.